File: Allin.Backend.Net/Microservices/Admin/Allin.Admin.Application/Models/BaseValueItemModel.cs
using Allin.Admin.Infrastructure.Persistence;
using Allin.Common.Models;
using Allin.Common.Utilities.Mappings;
using AutoMapper;

namespace Allin.Admin.Application.Models
{
    public class BaseValueItemModel : BaseHierarchyModel, IMapFrom<BaseValueItem, BaseValueItemModel>
    {
        public long BaseValueId { get; set; }
        public string Title { get; set; }
        public short Order { get; set; }
        public int Value { get; set; }
        public string Description { get; set; }

        // extra properties
        public string BaseValueUniqueName { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<BaseValueItem, BaseValueItemModel>()
                    .ForMember(dest => dest.BaseValueUniqueName, opt => opt.MapFrom(src => src.BaseValue.UniqueName))
                   ;
        }

    }
}

File: allin-front-angular/src/app/subsystems/admin/models/queries/base-value-item-model.ts
export interface BaseValueItemModel {
    id: number;
    parentId?: number;
    hierarchy: string;
    title: string;
    value: number;
    uniqueName: string;
    description: string;
    baseValueId: number;
    order: number;
    baseValueUniqueName: string;
    children: BaseValueItemModel[];
}
